### PR TITLE
FIX: Scheduler failed to run any jobs after Redis flush

### PR DESCRIPTION
**Pull request by gschlager**
This adds a check that runs every 60 seconds. It schedules all available jobs for a queue if the keys of the per host queue and the global queue are missing from Redis.

### CODE REVIEW

File: spec/mini_scheduler/manager_spec.rb [hunk 1]
```diff
@@ -195,6 +195,33 @@ def perform
...
+
+    it 'should recover from Redis flush' do
+      manager = MiniScheduler::Manager.new(enable_stats: false)
+      manager.ensure_schedule!(Testing::SuperLongJob)
+      manager.ensure_schedule!(Testing::PerHostJob)
+
+      expect(queued_jobs(manager, with_hostname: false)).to include(Testing::SuperLongJob)
+      expect(queued_jobs(manager, with_hostname: true)).to include(Testing::PerHostJob)
+
+      redis.flushdb
```
Comment (SamSaffron) [inline spec/mini_scheduler/manager_spec.rb:212]: flushdb kind of makes me nervous, is there any way you can be more surgical here? Maybe if all our keys are prefixed just delete keys with that prefix? do we flushdb much in our specs?

**Comment by SamSaffron** [comment]
Nice change and I support the naming change as well!

File: spec/mini_scheduler/manager_spec.rb (hunk 1 above)
Comment (gschlager) [reply spec/mini_scheduler/manager_spec.rb]: I think we flush a couple of times in Discourse core. Anyway, I pushed a new version which deletes only keys with the `_scheduler_` prefix.

**Comment by SamSaffron** [comment]
Looks good to me... merge away

**Review by SamSaffron — APPROVED** [review]

**Pull request merged**